code/modules/mob/living/carbon/human/life.dm

```diff
@@ -812,6 +827,37 @@
 		bodytemperature += recovery_amt
 
 	//This proc returns a number made up of the flags for body parts which you are protected on. (such as HEAD, UPPER_TORSO, LOWER_TORSO, etc. See setup.dm for the full list)
+/mob/living/carbon/human/proc/add_hypothermia(var/amount)
+	if(hypothermia+amount<=0)
+		hypothermia=0
+	else
+		if(hypothermia+amount>100)
+			hypothermia=100
+		hypothermia+=amount
+		if(hypothermia>5 && hypothermia<75)
+			if(prob(10))
+				say("*shiver")
+		if(hypothermia>30)
+			if(prob(50))
+				stuttering = hypothermia-25
+			if(hypothermia>50)
+				if(prob(40))
+					apply_effect(hypothermia-45,EYE_BLUR)
+				if(hypothermia>70)
+					if(prob(30))
+						Confuse(hypothermia-65)
+					if(hypothermia>80)
+						if(prob(15))
+							apply_effect(hypothermia-79,WEAKEN)

```

Comment:
Hmmm. Yea, I do feel like there should be a point at which the player needs someone to come get them(since in theory the final stage of hypothermia before death would be unconsciousness.) but I think I'll replace weaken with halloss.